packages/dds/tree/src/simple-tree/prepareForInsertion.ts

```diff
@@ -206,25 +205,32 @@ function walkMapTree(
 	}
 }
 
-function bindProxies(proxies: readonly RootedProxyPaths[], forest: IForestSubscription): void {
-	// Only subscribe to the event if there is at least one proxy tree to hydrate - this is not the case when inserting an empty array [].
-	if (proxies.length > 0) {
+/**
+ * Register a collection of nodes with the forest so that they can be hydrated.
+ * @param locatedNodes - the nodes to register with the forest.
+ * Each index in this array expects its content to be added and produce its own `afterRootFieldCreated` event.
+ * If array subsequence insertion is optimized to produce a single event, this will not work correctly as is, and will need to be modified to take in a single {@link LocatedNodesBatch}.
+ */
+function bindTreeNodes(
+	locatedNodes: readonly LocatedNodesBatch[],
+	forest: IForestSubscription,
+): void {
+	// Only subscribe to the event if there is at least one TreeNode tree to hydrate - this is not the case when inserting an empty array [].
+	if (locatedNodes.length > 0) {
 		// Creating a new array emits one event per element in the array, so listen to the event once for each element
 		let i = 0;
 		const off = forest.events.on("afterRootFieldCreated", (fieldKey) => {
 			// Non null asserting here because of the length check above
 			// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-			(proxies[i]!.rootPath as Mutable<UpPath>).parentField = fieldKey;
-			// Non null asserting here because of the length check above
-			// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-			for (const { path, proxy } of proxies[i]!.proxyPaths) {
-				getKernel(proxy).anchorProxy(forest.anchors, path);
+			const batch = locatedNodes[i]!;
```

Comment:
```suggestion
			const batch = locatedNodes[i] ?? oob();
```

Reply:
to get rid of the annoying lint comments.

Or maybe a `locatedNodes[i] ?? fail("Expected node")` is more appropriate, since technically this function can't guarantee with certainty that `locatedNodes` hasn't changed before the event is fired.

Reply:
I'll include this in my next cleanup PR, which should be ready shortly.